app/app.go

```diff
@@ -56,333 +77,314 @@ var (
+	bApp.SetParamStore(app.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramskeeper.ConsensusParamsKeyTable()))
 
-	app.SupplyKeeper = supply.NewKeeper(
-		app.cdc,
-		keys[supply.StoreKey],
-		app.AccountKeeper,
-		app.bankKeeper,
-		maccPerms,
+	app.AccountKeeper = authkeeper.NewAccountKeeper(
+		appCodec, keys[authtypes.StoreKey], app.GetSubspace(authtypes.ModuleName), authtypes.ProtoBaseAccount, maccPerms,
 	)
 
-	stakingKeeper := staking.NewKeeper(
-		app.cdc,
-		keys[staking.StoreKey],
-		app.SupplyKeeper,
-		app.subspaces[staking.ModuleName],
+	app.BankKeeper = bankkeeper.NewBaseKeeper(
+		appCodec, keys[banktypes.StoreKey], app.AccountKeeper, app.GetSubspace(banktypes.ModuleName), app.ModuleAccountAddrs(),
 	)
 
-	app.distrKeeper = distr.NewKeeper(app.cdc, keys[distr.StoreKey], app.subspaces[distr.ModuleName], &stakingKeeper,
-		app.SupplyKeeper, auth.FeeCollectorName, app.ModuleAccountAddrs())
-
-	app.slashingKeeper = slashing.NewKeeper(
-		app.cdc, keys[slashing.StoreKey], &stakingKeeper, app.subspaces[slashing.ModuleName],
+	stakingKeeper := stakingkeeper.NewKeeper(
+		appCodec, keys[stakingtypes.StoreKey], app.AccountKeeper, app.BankKeeper, app.GetSubspace(stakingtypes.ModuleName),
 	)
 
-	app.stakingKeeper = *stakingKeeper.SetHooks(
-		staking.NewMultiStakingHooks(app.distrKeeper.Hooks(), app.slashingKeeper.Hooks()))
-
-	app.OracleKeeper = oracle.NewKeeper(
-		app.cdc,
-		keys[oracle.StoreKey],
-		app.stakingKeeper,
-		oracle.DefaultConsensusNeeded,
+	app.DistrKeeper = distrkeeper.NewKeeper(
+		appCodec, keys[distrtypes.StoreKey], app.GetSubspace(distrtypes.ModuleName), app.AccountKeeper, app.BankKeeper,
+		&stakingKeeper, authtypes.FeeCollectorName, app.ModuleAccountAddrs(),
 	)
 
-	app.EthBridgeKeeper = ethbridge.NewKeeper(
-		app.cdc,
-		app.SupplyKeeper,
-		app.OracleKeeper,
-		keys[ethbridge.StoreKey],
+	app.SlashingKeeper = slashingkeeper.NewKeeper(
+		appCodec, keys[slashingtypes.StoreKey], &stakingKeeper, app.GetSubspace(slashingtypes.ModuleName),
 	)
 
-	app.clpKeeper = clp.NewKeeper(
-		app.cdc,
-		keys[clp.StoreKey],
-		app.bankKeeper,
-		app.SupplyKeeper,
-		app.subspaces[clp.ModuleName])
+	// register the staking hooks
+	// NOTE: stakingKeeper above is passed by reference, so that it will contain these hooks
+	app.StakingKeeper = *stakingKeeper.SetHooks(
+		stakingtypes.NewMultiStakingHooks(app.DistrKeeper.Hooks(), app.SlashingKeeper.Hooks()),
+	)
 
-	app.faucetKeeper = faucet.NewKeeper(
-		app.SupplyKeeper,
-		app.cdc,
-		keys[faucet.StoreKey],
-		app.bankKeeper)
+	// app.OracleKeeper = oracle.NewKeeper(
+	// 	appCodec,
+	// 	keys[oracle.StoreKey],
+	// 	app.StakingKeeper,
+	// 	oracle.DefaultConsensusNeeded,
+	// )
+
+	// app.EthBridgeKeeper = ethbridge.NewKeeper(
+	// 	appCodec,
+	// 	app.BankKeeper,
+	// 	app.OracleKeeper,
+	// 	keys[ethbridge.StoreKey],
+	// )
+
+	// app.clpKeeper = clp.NewKeeper(
+	// 	appCodec,
+	// 	keys[clp.StoreKey],
+	// 	app.BankKeeper,
+	// 	app.subspaces[clp.ModuleName])
```

Comment:
Same here, can we simply remove commented code?

Reply:
yea i can remove. I commented because it will be broght back later in subsequent PRs.